pilot/pkg/networking/util/internal_upstream.go

```diff
@@ -30,16 +30,28 @@ func RawBufferTransport() *core.TransportSocket {
 	}
 }
 
-// DefaultInternalUpstreamTransportSocket provides an internal_upstream transport that does not passthrough any metadata.
-var DefaultInternalUpstreamTransportSocket = &core.TransportSocket{
-	Name: "internal_upstream",
-	ConfigType: &core.TransportSocket_TypedConfig{TypedConfig: protoconv.MessageToAny(&internalupstream.InternalUpstreamTransport{
-		TransportSocket: RawBufferTransport(),
-	})},
+// InternalUpstreamTransportSocket wraps provided transport socket into Envoy InteralUpstreamTransport.
+func InternalUpstreamTransportSocket(name string, transport *core.TransportSocket) *core.TransportSocket {
+	return &core.TransportSocket{
+		Name: name,
+		ConfigType: &core.TransportSocket_TypedConfig{
+			TypedConfig: protoconv.MessageToAny(
+				&internalupstream.InternalUpstreamTransport{
+					TransportSocket: transport,
+				},
+			),
+		},
+	}
 }
 
+// DefaultInternalUpstreamTransportSocket provides an internal_upstream transport that does not passthrough any metadata.
+var DefaultInternalUpstreamTransportSocket = InternalUpstreamTransportSocket("internal_upstream", RawBufferTransport())
+
+// WaypointMetadataKey is the name of the dynamic metadata structure in Envoy containing authority for double-HBONE connections.
+const WaypointMetadataKey = "waypoint"
```

Comment:
maybe nit but given we may eventually want other sources do we want to make this more generic? Maybe re-using the existing `istio` key?